airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceOperations.kt

```diff
@@ -63,81 +61,70 @@ import jakarta.inject.Singleton
 class MysqlSourceOperations : JdbcMetadataQuerier.FieldTypeMapper, SelectQueryGenerator {
     override fun toFieldType(c: JdbcMetadataQuerier.ColumnMetadata): FieldType =
         when (val type = c.type) {
-            is SystemType -> leafType(c.type.typeName, type.scale != 0)
-            is UserDefinedArray -> ArrayFieldType(recursiveArrayType(type))
-            is UserDefinedType -> PokemonFieldType
+            is SystemType -> leafType(type)
+            else -> PokemonFieldType
         }
 
-    private fun recursiveArrayType(type: UserDefinedArray): JdbcFieldType<*> =
-        when (val elementType = type.elementType) {
-            is SystemType -> {
-                val leafType: JdbcFieldType<*> =
-                    leafType(elementType.typeName, elementType.scale != 0)
-                if (leafType == OffsetDateTimeFieldType) {
-                    // Mysql's JDBC driver has a bug which prevents object conversions in
-                    // ArrayDataResultSet instances. Fall back to strings instead.
-                    PokemonFieldType
+    private fun leafType(type: SystemType): JdbcFieldType<*> {
+        val typeName = type.typeName
```

Comment:
Is it possible to call `MysqlType.getByName(type.typeName)` to get the `MysqlType` enum value and dispatch on that instead?
The resulting `when` dispatch block would look a lot nicer that way.

Reply:
FYI, style nit: Type declarations are not optional when the type isn't obvious.

I'll look into having this as a linter rule, PRs are better off without these kinds of comments.

Reply:
nice. thanks!